gemrb/plugins/GUIScript/GUIScript.cpp

```diff
@@ -6689,17 +6693,20 @@ static PyObject* GemRB_TextArea_ListResources(PyObject * /*self*/, PyObject* arg
 			if (name[0] == '.' || dirit.IsDirectory() != dirs)
 				continue;
 
-			String* string = StringFromCString(name);
+			char * str = (char*) ConvertCharEncoding(name, core->SystemEncoding, core->TLKEncoding.encoding.c_str());
+			String* string = StringFromCString(str);
+
 			if (dirs == false) {
 				size_t pos = string->find_last_of(L'.');
 				if (pos == String::npos || (type == DIRECTORY_CHR_SOUNDS && pos-- == 0)) {
+					free(str);
 					delete string;
 					continue;
 				}
 				string->resize(pos);
 			}
-			StringToUpper(*string);
 			strings.push_back(*string);
+			free(str);
```

Comment:
You can just move these two frees to right after `string` is created, since they're not used later any more.